plugins/vli/fu-vli-pd-device.c

```diff
@@ -348,36 +349,80 @@ fu_vli_pd_device_setup(FuDevice *device, GError **error)
 
 	/* get device kind if not already in ROM mode */
 	if (fu_vli_device_get_kind(FU_VLI_DEVICE(self)) == FU_VLI_DEVICE_KIND_UNKNOWN) {
-		if (!fu_vli_pd_device_read_reg(self, 0x0018, &tmp, error))
-			return FALSE;
-		switch (tmp & 0xF0) {
-		case 0x00:
-			fu_vli_device_set_kind(FU_VLI_DEVICE(self), FU_VLI_DEVICE_KIND_VL100);
-			break;
-		case 0x10:
-			/* this is also the code for VL101, but VL102 is more likely */
-			fu_vli_device_set_kind(FU_VLI_DEVICE(self), FU_VLI_DEVICE_KIND_VL102);
-			break;
-		case 0x80:
-			fu_vli_device_set_kind(FU_VLI_DEVICE(self), FU_VLI_DEVICE_KIND_VL103);
-			break;
-		case 0x90:
-			fu_vli_device_set_kind(FU_VLI_DEVICE(self), FU_VLI_DEVICE_KIND_VL104);
-			break;
-		default:
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_INVALID_FILE,
-				    "unable to map 0x0018=0x%02X to device kind",
-				    tmp);
+		if (!fu_vli_pd_device_read_reg(self, 0x0018, &value, error))
```

Comment:
what's 0x0018?

Reply:
Sherlock will check.

Reply:
Can I use
`#define FU_VLI_PD_REGISTER_ADDRESS_PROJ_LEGACY   0x0018` ?
(0x0018 is old addrss of project ID, 0x009C 0x009D are new)

Reply:
works for me!